content/en/blog/general/introducing-chainsaw/index.md

```diff
@@ -0,0 +1,339 @@
+
+> Operators are software extensions to Kubernetes that make use of custom resources to manage applications and their components. Operators follow Kubernetes principles, notably the control loop.
+
+They often rely on Custom Resource Definitions and continuously reconcile the cluster state with the spec of Custom Resources.
+
+## How do we test a Kubernetes operator ?
+
+An operator is essentially responsible for watching certain resources in a cluster and react to maintain a state matching the spec described in the Custom Resources.
+
+Testing an operator boils down to creating, updating or deleting certain resources and veriify the state of the cluster changes accordingly.
+
+
+For example, an operator could be responsible for managing role bindings and service accounts in a cluster based on a simplified definition of permissions. This operator exists, see [rbac-manager](https://github.com/FairwindsOps/rbac-manager) from FairWinds.
+
+In the next sections of this blog post I will demonstrate how Chainsaw can help testing the [rbac-manager](https://github.com/FairwindsOps/rbac-manager) operator.
+
+## Getting started
+
+Before we can look at Chainsaw we need a Kubernetes cluster with `rbac-manager` installed.
+We can create a local cluster with [KinD](https://kind.sigs.k8s.io) and use [Helm](https://helm.sh) to install the operator.
+
+```sh
+# create a cluster
+$ kind create cluster
+
+# deploy rbac-manager
+$ helm install rbac-manager --repo https://charts.fairwinds.com/stable rbac-manager --namespace rbac-manager --create-namespace
+```
+
+Once the operator installed, you should see a new Custom Resource Definition in the cluster:
+
+```sh
+$ kubectl get crd
+
+NAME                                         CREATED AT
+rbacdefinitions.rbacmanager.reactiveops.io   2023-12-12T12:20:19Z
+```
+
+## Install Chainsaw
+
+Chainsaw can be installed in [different ways](https://kyverno.github.io/chainsaw/latest/install/). If you are using MacOS or Linux, the simplest solution is to use [brew](https://brew.sh/).
+
+```sh
+# add the chainsaw tap
+$ brew tap kyverno/chainsaw https://github.com/kyverno/chainsaw
+
+# install chainsaw
+$ brew install kyverno/chainsaw/chainsaw
+```
+
+## What is a test ?
+
+To put it simply, a test can be represented as **an ordered sequence of test steps**.
+
+Test steps within a test are run sequentially: **if any of the test steps fail, the entire test is considered failed**.
+
+A test step can consist in one or more operations:
+
+- To delete resources present in a cluster
+- To create or update resources in a cluster
+- To assert one or more resources in a cluster meet the expectations (or the opposite)
+- To run arbitrary commands or scripts
+
+In Chainsaw, tests are entirely declarative and created with yaml files.
+
+## Our first test
+
+In this first test, we're going to create an `RBACDefinition` and verify the `rbac-manager` operator created the corresponding `ClusterRoleBinding` in the cluster.
```

Comment:
```suggestion
In this first test, we're going to create an `RBACDefinition` and verify the `rbac-manager` operator created the corresponding ClusterRoleBinding in the cluster.
```

You don't stylize Kubernetes resources in code formatting so be consistent.

Reply:
And if you're going to write this as a formal resource (ClusterRoleBinding and not cluster role binding) you should be consistent there as well.